### PR TITLE
Add `files_for_e3sm` steps for ocean and sea-ice meshes

### LINKED ISSUE #569: Add ocean and sea-ice meshes (without initial conditions) to `files_for_e3sm`
These should be staged in `inputdata/shared/meshes/mpas/ocean` and `inputdata/shared/meshes/mpas/seaice`, respectively.

### PR DESCRIPTION
**Pull request by xylar**
These also need to be added to `inputdata` in `shared/meshes/mpas`

This merge also makes public some functions from the `compass.ocean.vertical` module that are used to add `cellMask` and `zMid` variables to the MPAS-Ocean mesh.

Checklist
* [x] User's Guide has been updated
* [x] Developer's Guide has been updated
* [x] API documentation in the Developer's Guide (`api.rst`) has any new or modified class, method and/or functions listed
* [x] Documentation has been [built locally](https://mpas-dev.github.io/compass/latest/developers_guide/building_docs.html) and changes look as expected
* [x] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes

closes #569

### CODE REVIEW

**Comment by xylar** [comment]
## Testing

I tested this on the following 6 meshes:
* EC30to60E2r2
* EC30to60E2r3
* ECwISC30to60E2r1
* SOwISC12to60E2r4
* WC14to60E2r3
* WC14to60E2r5

which I have created and placed in:
```
/lcrc/group/e3sm/public_html/inputdata/share/meshes/mpas
```

**Comment by xylar** [comment]
Based off of #563, so in draft mode until that gets merged.

**Comment by xylar** [comment]
Can't build docs locally until I fix #567

**Comment by xylar** [comment]
@mark-petersen, I could use your review on this.  I'll point out specific things I'd appreciate your feedback on.

File: compass/ocean/tests/global_ocean/files_for_e3sm/ocean_mesh.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+        dest_filename = f'{self.mesh_short_name}.{self.creation_date}.nc'
+
+        with xr.open_dataset('initial_state.nc') as ds:
+
+            keep_vars = self.mesh_vars + [
+                'refBottomDepth', 'vertCoordMovementWeights',
+                'bottomDepth', 'maxLevelCell',
+                'layerThickness', 'restingThickness'
+            ]
```
Comment (xylar) [inline compass/ocean/tests/global_ocean/files_for_e3sm/ocean_mesh.py:53]: @mark-petersen, in addition to the horizontal mesh variables, these seem to be the minimum set available for defining the vertical grid.  Does that seem right?

File: compass/ocean/tests/global_ocean/files_for_e3sm/files_for_e3sm_step.py [hunk 2]
```diff
@@ -78,12 +87,37 @@ def __init__(self, test_case, name, subdir=None, cpus_per_task=1,
...
+        self.mesh_vars = [
+            'areaCell', 'cellsOnCell', 'edgesOnCell', 'indexToCellID',
+            'latCell', 'lonCell', 'meshDensity', 'nEdgesOnCell',
+            'verticesOnCell', 'xCell', 'yCell', 'zCell', 'angleEdge',
+            'cellsOnEdge', 'dcEdge', 'dvEdge', 'edgesOnEdge',
+            'indexToEdgeID', 'latEdge', 'lonEdge', 'nEdgesOnCell',
+            'nEdgesOnEdge', 'verticesOnEdge', 'weightsOnEdge', 'xEdge',
+            'yEdge', 'zEdge', 'areaTriangle', 'cellsOnVertex', 'edgesOnVertex',
+            'indexToVertexID', 'kiteAreasOnVertex', 'latVertex',
+            'lonVertex', 'xVertex', 'yVertex', 'zVertex']
```
[the remark this answers is not in the document]
Comment (mark-petersen) [reply compass/ocean/tests/global_ocean/files_for_e3sm/files_for_e3sm_step.py:105]: For reference, these are my notes on how to add a mesh for a paraview visualiztion from a netcdf file without a mesh:
```
set GridVars=latCell,lonCell,maxLevelCell,nEdgesOnCell,xCell,yCell,zCell,dcEdge,dvEdge,indexToEdgeID,latEdge,lonEdge,nEdgesOnEdge,xEdge,yEdge,zEdge,areaTriangle,indexToVertexID,latVertex,lonVertex,xVertex,yVertex,zVertex,edgesOnVertex,cellsOnVertex,indexToCellID,indexToEdgeID,indexToVertexID,cellsOnEdge,edgesOnCell,edgesOnEdge,cellsOnCell,refBottomDepth,verticesOnCell
ncks -d Time,0 -v ${GridVars},temperature file1.nc file2.nc
```
Yes, your list includes all the horizontal variables from my notes. I agree that you have them all.

File: compass/ocean/tests/global_ocean/files_for_e3sm/ocean_mesh.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                'done on cells.'
+            ds['ssh'] = compute_ssh_from_layer_thickness(
+                ds.layerThickness, ds.bottomDepth, ds.cellMask)
+            ds.ssh.attrs['units'] = 'm'
+            ds.ssh.attrs['long_name'] = 'sea surface height'
+            ds['zMid'] = compute_zmid_from_layer_thickness(
+                ds.layerThickness, ds.ssh, ds.cellMask)
+            ds.zMid.attrs['units'] = 'm'
+            ds.zMid.attrs['long_name'] = \
+                'z-coordinate of the mid-depth of the layer'
```
[the remark this answers is not in the document]
Comment (mark-petersen) [reply compass/ocean/tests/global_ocean/files_for_e3sm/ocean_mesh.py:110]: Yes, that looks good.  The variables edgeMask and vertexMask would match your cellMask, but we probably don't need them.

[the remark this answers is not in the document]
Comment (xylar) [reply compass/ocean/tests/global_ocean/files_for_e3sm/ocean_mesh.py:110]: I thought about that, too.  I don't currently have compass routines for computing those (and I needed `cellMask` for computing `ssh` and `zMid`) so I decided to let them be.

Thanks for taking a look!

**Review by mark-petersen — APPROVED** [review]
Yes, please merge. Looks good, thank you!

**Pull request merged**